device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -763,9 +763,14 @@ public void setProxySettings(ProxySettings proxySettings)
         }
 
         IotHubClientProtocol protocol = this.deviceIO.getProtocol();
-        if (protocol != HTTPS)
+        if (protocol != HTTPS && protocol != AMQPS_WS)
         {
-            throw new UnsupportedOperationException("Cannot use proxy unsupported unless you are using HTTPS");
+            throw new UnsupportedOperationException("Use of proxies is unsupported unless using HTTPS or AMQPS_WS");
```

Comment:
It should be `IllegalArgumentException` instead of `UnsupportedOperationException`.

Reply:
I'll address this in the next PR, sure